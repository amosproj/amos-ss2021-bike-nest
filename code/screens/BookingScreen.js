import React, { useEffect } from 'react';
import { Pressable, StyleSheet, Text, TextInput, View, Image} from 'react-native';
import { Dimensions } from "react-native";
import Colors from '../styles/Colors';
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';
import { mainStyles } from "../styles/MainStyles";
import ModalDropdown from 'react-native-modal-dropdown';
import colors from '../styles/Colors';
import { BikenestService } from "../services/BikenestService";
import { Alert } from 'react-native';

var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height

export default function BookingScreen({ route, navigation }) {
  let bikenestService = new BikenestService();

  const [textSlots, setTextSlots] = React.useState();
  const [textHours, setTextHours] = React.useState();
  const [textEbike, setTextEbike] = React.useState();
  const [costSlots, setcostSlots] = React.useState();
  const [costHours, setcostHours] = React.useState();
  const [costEbike, setcostEbike] = React.useState();
  const [estimatedPrice, setEstimatedPrice] = React.useState();

  const [spotsLeft, setSpotsLeft] = React.useState(0);
  const [chargingOption, setChargingOption] = React.useState(true);
  const [bikenestName, setbikenestName] = React.useState('default Bikenest');

  let bikenest = route.params.state;
  console.log(bikenest.id);

  const fetchBikenestInfos = () => {
      bikenestService.getBikenestInfo(bikenest.id).then((response) => {
        setSpotsLeft(response.currentSpots);
        setChargingOption(response.chargingAvailable);
        setbikenestName(response.name);
      }).catch((error) => {
        console.error("getBikenest Infos Error:" + error);
     });
  };

  const onPressInfo = () => {
    //zurück zu Find Bike Nest
    alert('this is an info.');
  };
  const onPressOrder = () => {
    //weiter zu order verarbeitung
    if(textSlots != null && textHours != null && textEbike != null){
      navigation.navigate("Payment", {state: bikenest, name: bikenestName, slots: textSlots, time: textHours, ebike: textEbike, price: estimatedPrice});
    } else {
      Alert.alert("Halt", "Bitte wähle Slots, Stunden und Ebike-Option aus")
    } 
  }
  const getLocation = () => {
    fetchBikenestInfos();
    return bikenestName;
  };
  const getSelectedHours = (value) => {
    if(value == 0){
      setTextHours('1 Tag');
      setcostHours(1);
    }else if(value == 1){
      setTextHours('2 Tage');
      setcostHours(2);
    }else if(value == 2){
      setTextHours('3 Tage');
      setcostHours(3);
    }else if(value == 3){
      setTextHours('1 Monat');
      setcostHours(12);
    } else {
      setTextHours('Keine Zeit ausgewählt.');
      setcostHours(0);
    }
  }
  const getSelectedSlots = (value) => {

    if(value == 0){
      setTextSlots('1 Slot');
      setcostSlots(1);
    } else if(value == 1){
      setTextSlots('2 Slots');
      setcostSlots(2);
    } else {
      setTextSlots('Keine Slots ausgewählt.');
      setcostSlots(0);
    }
  }

  const getSelectedEbike = (value) => {
    if(value == 0){
      setTextEbike('1 E-Bike Ladestation');
      setcostEbike(5);
    } else if(value == 1){
      setTextEbike('2 E-Bike Ladestation');
      setcostEbike(10);
    } else if(value == 2){
      setTextEbike('keine E-Bike Ladestation');
      setcostEbike(0);
    } else {
      setTextEbike('kein E-bike ausgewählt.');
      setcostEbike(0);
    }
  }

  useEffect(()=>{
    let price = 0

    if(costHours != null){
      price = costHours;
    } 
    if (costHours != null && costSlots != null){
      price = (costHours*costSlots);
    } 
    if (costHours != null && costSlots != null && costEbike != null){
      price = (costHours*costSlots) + costEbike;
    }
    setEstimatedPrice(price);
     
  });
  return (
    <View style={myStyles.container}>
      <View style={myStyles.paymentContainer}>
        <View style={{alignSelf:'center'}}>
            <Text style={myStyles.h2}>
                Buchungsinformationen <Image onPress={() => onPressInfo()} source={require('../assets/payment/info.png')}/>
            </Text>
        </View>
        <View style={myStyles.headline}>
            <Text style={myStyles.h3}> Details </Text>
            <Text style={[myStyles.h3, {fontWeight: 'bold'}]} onPress={()=>navigation.navigate("FindBikeNest")}> Ändern </Text>
        </View>
        <View style={myStyles.headline}>
            <Text style={myStyles.stdText}>
              BIKE NEST {"\n"}
              {getLocation()}
            </Text>
        </View>
        <Image source={require('../assets/payment/Divider.png')} style={myStyles.divider}/>
        <View style={myStyles.headline}>
            <Text style={myStyles.h3}> {"\n"} Zusätzliche Informationen </Text>
        </View>
       
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Wie viele Slots? </Text>
          <ModalDropdown onSelect={(value) => getSelectedSlots(value)} options={['1 Slot', '2 Slots']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Wie lange buchen? </Text>
          <ModalDropdown onSelect={(value) => getSelectedHours(value)} options={['1 Tag', '2 Tage', '3 Tage', '1 Monat']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> E-Bike Ladestation? </Text>
          <ModalDropdown onSelect={(value) => getSelectedEbike(value)} options={['ja, eine', 'ja, zwei', 'nein, danke']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>

        <Image source={require('../assets/payment/Divider.png')} style={myStyles.divider}/>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Zusammenfassung</Text>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.stdText}>
                {"\n"} 
                BIKE NEST {"\n"}
                {getLocation()} {"\n"}
                {textSlots}{"\n"}
                {textHours}{"\n"}
                {textEbike}{"\n"}
                ~{estimatedPrice}€
              </Text>
        </View>
          {/* <Pressable style={[myStyles.reserved, {justifyContent: 'flex-end'}]}  onPress={() => getPrice(this)}>
            <Text style={myStyles.h3}>Preis Kalkulieren</Text> 
          </Pressable> */}
          <Pressable style={[myStyles.reserved, {justifyContent: 'flex-end'}]}  onPress={() => onPressOrder(this)}>
            <Text style={myStyles.h3}>Weiter</Text> 
            <Image style={{margin: 10}} source={require('../assets/payment/mail-send.png')} />
          </Pressable>
      </View>
      <BikeNest_NavigationFooter/>
    </View>
  )
}

const myStyles = StyleSheet.create({
  container: {
      flex: 1,
      backgroundColor: '#ffffff',
  },
  paymentContainer: {
    flex: 1,
    padding: 15,
    marginTop: 30,
  },
  h2: {
      fontSize: 27,
      letterSpacing: 2,
      fontWeight: "400",
  },
  h3:{
      fontSize: 18,
      fontWeight: "300",
  },
  stdText:{
      fontSize: 14,
  },
  divider: {
      marginLeft: -15,
      marginTop: 10,
      marginBottom: 10,
      width: 500,
  },
  headline: {
      flex: 1,
      fontSize: 18,
      fontWeight: "300",
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center', 
  },
  reserved: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center', 
  },
  zahlungsmethode: {
      flexDirection: 'column',
      justifyContent: 'space-between',
      alignItems: 'center', 
  },
  button: {
      alignItems: 'center',
      justifyContent: 'center',
      elevation: 3,
      width: 328,
      height: 55,
      borderRadius: 38,
      margin: 9,
      backgroundColor: Colors.UI_Light_4
  },
  btnPress: {
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 3,
    width: 328,
    height: 55,
    borderRadius: 38,
    margin: 9,
    backgroundColor: Colors.UI_Light_4,
    borderColor: Colors.UI_Light_3,
  },
  buttonContent: {
      flexDirection: 'row',
      justifyContent: 'space-evenly',
      width: '100%'
    },
  buttonImage: {
      alignSelf: 'center'
    },
  halfButton: {
      width: 150,
      height: 55,
      color: Colors.UI_Light_3,
      backgroundColor: Colors.UI_Light_4,
      borderRadius: 38,
      margin: 10,
      padding: 15,
    }
})
