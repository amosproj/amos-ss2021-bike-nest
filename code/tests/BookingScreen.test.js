import React from 'react';
import renderer from 'react-test-renderer';
import BookingScreen from '../screens/BookingScreen';

it('renders correctly', () => {
    const params = { state: jest.fn(), nestID: jest.fn() };
    const tree = renderer
        .create(<BookingScreen route={{params}}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});