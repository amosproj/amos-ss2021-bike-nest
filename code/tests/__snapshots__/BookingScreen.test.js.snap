// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 30,
        "padding": 15,
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 27,
            "fontWeight": "400",
            "letterSpacing": 2,
          }
        }
      >
        Buchungsinformationen 
        <Image
          onPress={[Function]}
          source={1}
        />
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         Details 
      </Text>
      <Text
        onPress={[Function]}
        style={
          Array [
            Object {
              "fontSize": 18,
              "fontWeight": "300",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
         Ändern 
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        BIKE NEST 
        

        default Bikenest
      </Text>
    </View>
    <Image
      source={1}
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": -15,
          "marginTop": 10,
          "width": 500,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         
        

         Zusätzliche Informationen 
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         Wie viele Slots? 
      </Text>
      <View
        animated={true}
        defaultIndex={-1}
        defaultValue="Bitte wählen..."
        disabled={false}
        dropdownTextStyle={
          Object {
            "fontSize": 18,
          }
        }
        isFullWidth={false}
        keyboardShouldPersistTaps="never"
        multipleSelect={false}
        onSelect={[Function]}
        options={
          Array [
            "1 Slot",
            "2 Slots",
          ]
        }
        renderButtonComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        renderRightComponent={[Function]}
        renderRowComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        scrollEnabled={true}
        showsVerticalScrollIndicator={true}
        textStyle={
          Object {
            "fontSize": 18,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Bitte wählen...
            </Text>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         Wie lange buchen? 
      </Text>
      <View
        animated={true}
        defaultIndex={-1}
        defaultValue="Bitte wählen..."
        disabled={false}
        dropdownTextStyle={
          Object {
            "fontSize": 18,
          }
        }
        isFullWidth={false}
        keyboardShouldPersistTaps="never"
        multipleSelect={false}
        onSelect={[Function]}
        options={
          Array [
            "1 Tag",
            "2 Tage",
            "3 Tage",
            "1 Monat",
          ]
        }
        renderButtonComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        renderRightComponent={[Function]}
        renderRowComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        scrollEnabled={true}
        showsVerticalScrollIndicator={true}
        textStyle={
          Object {
            "fontSize": 18,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Bitte wählen...
            </Text>
            <View />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         E-Bike Ladestation? 
      </Text>
      <View
        animated={true}
        defaultIndex={-1}
        defaultValue="Bitte wählen..."
        disabled={false}
        dropdownTextStyle={
          Object {
            "fontSize": 18,
          }
        }
        isFullWidth={false}
        keyboardShouldPersistTaps="never"
        multipleSelect={false}
        onSelect={[Function]}
        options={
          Array [
            "ja, eine",
            "ja, zwei",
            "nein, danke",
          ]
        }
        renderButtonComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        renderRightComponent={[Function]}
        renderRowComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        scrollEnabled={true}
        showsVerticalScrollIndicator={true}
        textStyle={
          Object {
            "fontSize": 18,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "fontSize": 18,
                    },
                    undefined,
                  ],
                ]
              }
            >
              Bitte wählen...
            </Text>
            <View />
          </View>
        </View>
      </View>
    </View>
    <Image
      source={1}
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": -15,
          "marginTop": 10,
          "width": 500,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
         Zusammenfassung
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "300",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        

        BIKE NEST 
        

        default Bikenest
         
        

        

        

        

        ~
        €
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "300",
          }
        }
      >
        Weiter
      </Text>
      <Image
        source={1}
        style={
          Object {
            "margin": 10,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Profil
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          (un)lock
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Navigation
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F28D35",
          "flex": 1,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          style={
            Object {
              "margin": 5,
            }
          }
        >
          <Image
            source={1}
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2F2F2",
              "fontSize": 14,
              "height": "100%",
              "textAlign": "center",
              "width": "100%",
            }
          }
        >
          Verlauf
        </Text>
      </View>
    </View>
  </View>
</View>
`;
